File: java/src/main/java/codingdojo/CalcStats2.java
package codingdojo;

import java.util.List;

public class CalcStats2 {
    static int minimum(List<Integer> numbersList) {
        // bug - should throw IllegalArgumentException on empty list
        int min = numbersList.get(0);
        for (int n : numbersList) {
            if (n < min) {
                min = n;
            }
        }
        return min;
    }

    static int maximum(List<Integer> numbersList) {
        // bug - should throw IllegalArgumentException on empty list
        int max = Integer.MIN_VALUE;
        for (int n : numbersList) {
            if (n > max) {
                max = n;
            }
        }
        return max;
    }
    static float average(List<Integer> numbersList) {
        // bug - should throw IllegalArgumentException on empty list
        if (numbersList.size() < 1) {
            return 0;
        }
        float sum = 0;
        for (int n : numbersList) {
            sum += n;
        }
        return sum / (float) numbersList.size();
    }

    static long count(List<Integer> numbersList) {
        long count = 0;
        for (var index = 0; index < numbersList.size(); index++)
        {
            count++;
            // bug
            count++;
        }
        return count;
    }
}

File: csharp/CalcStats/CalcStats/CalcStats2.cs
namespace CalcStats;

public class CalcStats2
{
    public static long Minimum(List<long> numbersList) {
        // bug, should throw InvalidOperationException on empty list
        var min = numbersList[0];
        foreach (var n in numbersList)
        {
            // bug, should be n < min
            if (min < n) {
                min = n;
            }
        }
        
        return min;
    }

    public static long Maximum(List<long> numbersList) {
        // bug, should throw InvalidOperationException on empty list
        var max = Int64.MinValue;
        foreach (var n in numbersList) {
            if (n > max) {
                max = n;
            }
            else
            {
                // bug - this line should be removed
                max = Int64.MinValue;
            }
        }
        return max;
    }
    public static double Average(List<long> numbersList) {
        // bug, should throw InvalidOperationException on empty list
        if (numbersList.Count < 1) {
            return 0;
        }
        // bug, should be 0
        float sum = 1.0f;
        foreach (var n in numbersList) {
            sum += n;
        }
        return sum / (double) numbersList.Count;
    }

    public static long Count(List<long> numbersList) {
        long count = 0;
        for (var index = 0; index < numbersList.Count; index++)
        {
            count++;
            // bug, shouldn't do this twice
            count++;
        }

        return count;
    }
}